From 78fd00072ca870afcc722c28bb731bff7b00af2f Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Mon, 4 Mar 2024 13:50:23 +0100
Subject: [PATCH] backport drm_unlocked_ioctl_flag_present changes from
 470.239.06

---
 conftest.sh | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/conftest.sh b/conftest.sh
index ce2d568..2ee2f50 100755
--- a/conftest.sh
+++ b/conftest.sh
@@ -6396,10 +6396,16 @@ compile_test() {
             # DRM_UNLOCKED was removed by commit 2798ffcc1d6a ("drm: Remove
             # locking for legacy ioctls and DRM_UNLOCKED") in Linux
             # next-20231208.
+            #
+            # DRM_UNLOCKED definition was moved from drmP.h to drm_ioctl.h by
+            # commit 2640981f3600 ("drm: document drm_ioctl.[hc]") in v4.12.
             CODE="
             #if defined(NV_DRM_DRM_IOCTL_H_PRESENT)
             #include <drm/drm_ioctl.h>
             #endif
+            #if defined(NV_DRM_DRMP_H_PRESENT)
+            #include <drm/drmP.h>
+            #endif
             int flags = DRM_UNLOCKED;"
 
             compile_check_conftest "$CODE" "NV_DRM_UNLOCKED_IOCTL_FLAG_PRESENT" "" "types"
-- 
2.20.1

