From 2d8337bb81bcc74dccae22ae2d70fc12d407f12d Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Tue, 11 Feb 2025 13:15:26 +0100
Subject: [PATCH] backport warning fixes from 535.43.02

---
 nvidia/nv-acpi.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/nvidia/nv-acpi.c b/nvidia/nv-acpi.c
index 4ef0c228..c20f4231 100644
--- a/nvidia/nv-acpi.c
+++ b/nvidia/nv-acpi.c
@@ -98,9 +98,6 @@ static int nv_acpi_notifier_call_chain_handler(
 )
 {
     struct acpi_bus_event *info = data;
-    nv_stack_t *sp = NULL;
-    nv_linux_state_t *nvl = container_of(nb, nv_linux_state_t, acpi_nb);
-    nv_state_t *nv = NV_STATE_PTR(nvl);
 
     /*
      * The ACPI_VIDEO_NOTIFY_PROBE will be sent for display hot-plug/unplug.
-- 
2.39.5

