From 153559b7b46a2dd0d9a7f76847fdf2f9e5e6e54c Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Sun, 8 Dec 2024 22:13:22 +0100
Subject: [PATCH] fix more warnings

---
 common/inc/nv-linux.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/common/inc/nv-linux.h b/common/inc/nv-linux.h
index 12fcf6ce..f22418ed 100644
--- a/common/inc/nv-linux.h
+++ b/common/inc/nv-linux.h
@@ -571,21 +571,27 @@ static NvBool nv_numa_node_has_memory(int node_id)
     { \
         (ptr) = kmalloc(size, NV_GFP_KERNEL); \
         if (ptr) \
+        { \
             NV_MEMDBG_ADD(ptr, size); \
+        } \
     }
 
 #define NV_KZALLOC(ptr, size) \
     { \
         (ptr) = kzalloc(size, NV_GFP_KERNEL); \
         if (ptr) \
+        { \
             NV_MEMDBG_ADD(ptr, size); \
+        } \
     }
 
 #define NV_KMALLOC_ATOMIC(ptr, size) \
     { \
         (ptr) = kmalloc(size, NV_GFP_ATOMIC); \
         if (ptr) \
+        { \
             NV_MEMDBG_ADD(ptr, size); \
+        } \
     }
 
 #if defined(__GFP_RETRY_MAYFAIL)
@@ -600,7 +606,9 @@ static NvBool nv_numa_node_has_memory(int node_id)
     { \
         (ptr) = kmalloc(size, NV_GFP_NO_OOM); \
         if (ptr) \
+        { \
             NV_MEMDBG_ADD(ptr, size); \
+        } \
     }
 
 #define NV_KFREE(ptr, size) \
-- 
2.39.5

