From 6be29e3d16a07c1e798d10717e8a6f94e4313389 Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Tue, 10 Dec 2024 20:08:50 +0100
Subject: [PATCH] fix more uvm warnings

---
 nvidia-uvm/uvm_tools.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/nvidia-uvm/uvm_tools.c b/nvidia-uvm/uvm_tools.c
index c7f92e43..7c0fee83 100644
--- a/nvidia-uvm/uvm_tools.c
+++ b/nvidia-uvm/uvm_tools.c
@@ -36,6 +36,7 @@
 #include "uvm_range_group.h"
 #include "uvm_mem.h"
 #include "nv_speculation_barrier.h"
+#include "uvm_tools_init.h"
 
 // We limit the number of times a page can be retained by the kernel
 // to prevent the user from maliciously passing UVM tools the same page
-- 
2.39.5

