From a2e458de3a650d77f44d7b7d0519f0a52c552893 Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Fri, 21 Feb 2025 10:21:42 +0100
Subject: [PATCH] backport module_import_ns_takes_constant changes from
 570.86.16

---
 conftest.sh          | 17 +++++++++++++++++
 nvidia/nvidia.Kbuild |  1 +
 2 files changed, 18 insertions(+)

diff --git a/conftest.sh b/conftest.sh
index 944229b2..7bdbc48e 100755
--- a/conftest.sh
+++ b/conftest.sh
@@ -6418,6 +6418,23 @@ compile_test() {
             compile_check_conftest "$CODE" "NV_DRM_UNLOCKED_IOCTL_FLAG_PRESENT" "" "types"
         ;;
 
+        module_import_ns_takes_constant)
+            #
+            # Determine if the MODULE_IMPORT_NS macro takes a string literal
+            # or constant.
+            #
+            # Commit cdd30ebb1b9f ("module: Convert symbol namespace to
+            # string literal") changed MODULE_IMPORT_NS to take a string
+            # literal in Linux kernel v6.13.
+            #
+            CODE="
+            #include <linux/module.h>
+
+            MODULE_IMPORT_NS(DMA_BUF);"
+
+            compile_check_conftest "$CODE" "NV_MODULE_IMPORT_NS_TAKES_CONSTANT" "" "generic"
+        ;;
+
         # When adding a new conftest entry, please use the correct format for
         # specifying the relevant upstream Linux kernel commit.
         #
diff --git a/nvidia/nvidia.Kbuild b/nvidia/nvidia.Kbuild
index 30f5e147..de97c7d8 100644
--- a/nvidia/nvidia.Kbuild
+++ b/nvidia/nvidia.Kbuild
@@ -264,3 +264,4 @@ NV_CONFTEST_GENERIC_COMPILE_TESTS += mdev_available
 NV_CONFTEST_GENERIC_COMPILE_TESTS += cmd_uphy_display_port_init
 NV_CONFTEST_GENERIC_COMPILE_TESTS += cmd_uphy_display_port_off
 NV_CONFTEST_GENERIC_COMPILE_TESTS += memory_failure_mf_sw_simulated_defined
+NV_CONFTEST_GENERIC_COMPILE_TESTS += module_import_ns_takes_constant
-- 
2.39.5

