From 53c8e33b014c64a394e6a20b1a293441ebfbc664 Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Thu, 29 Jan 2026 13:30:59 +0100
Subject: [PATCH] backport conftest.sh comment changes from 545.23.06

---
 conftest.sh | 479 +++++++++++++++++++++++++++-------------------------
 1 file changed, 253 insertions(+), 226 deletions(-)

diff --git a/conftest.sh b/conftest.sh
index 2e66fc24..0b43a6f2 100755
--- a/conftest.sh
+++ b/conftest.sh
@@ -549,10 +549,8 @@ compile_test() {
             # Determine if the set_pages_array_uc() function is present.
             # It does not exist on all architectures.
             #
-            # set_pages_array_uc() was added by commit
-            # 0f3507555f6fa4acbc85a646d6e8766230db38fc ("x86, CPA: Add
-            # set_pages_arrayuc and set_pages_array_wb") in v2.6.30-rc1 (Thu Mar
-            # 19 14:51:15 2009)
+            # Added by commit 0f3507555f6f ("x86, CPA: Add set_pages_arrayuc
+            # and set_pages_array_wb") in v2.6.30.
             #
             CODE="
             #include <linux/types.h>
@@ -597,8 +595,8 @@ compile_test() {
             #
             # Added by commit 3c299dc22635 ("PCI: add
             # pci_get_domain_bus_and_slot function") in 2.6.33 but aarch64
-            # support was added by commit d1e6dc91b532 
-            # ("arm64: Add architectural support for PCI") in 3.18-rc1
+            # support was added by commit d1e6dc91b532 ("arm64: Add
+            # architectural support for PCI") in 3.18.
             #
             CODE="
             #include <linux/pci.h>
@@ -1464,9 +1462,8 @@ compile_test() {
             #
             # Determine if the pde_data() function is present.
             #
-            # The commit c28198889c15 removed the function
-            # 'PDE_DATA()', and replaced it with 'pde_data()'
-            # ("proc: remove PDE_DATA() completely") in v5.17-rc1.
+            # PDE_DATA() was replaced with pde_data() by commit 359745d78351
+            # ("proc: remove PDE_DATA() completely") in v5.17.
             #
             CODE="
             #include <linux/proc_fs.h>
@@ -1607,8 +1604,8 @@ compile_test() {
             # based implementation") in v4.5
             #
             # Commit 0a0f0d8be76d ("dma-mapping: split <linux/dma-mapping.h>")
-            # in v5.10-rc1 (2020-09-22), moved get_dma_ops() function
-            # prototype from <linux/dma-mapping.h> to <linux/dma-map-ops.h>.
+            # in v5.10 moved get_dma_ops() function prototype from
+            # <linux/dma-mapping.h> to <linux/dma-map-ops.h>.
             #
             CODE="
             #if defined(NV_LINUX_DMA_MAP_OPS_H_PRESENT)
@@ -1778,9 +1775,8 @@ compile_test() {
 
         kernel_write_has_pointer_pos_arg)
             #
-            # Determine the pos argument type, which was changed by
-            # commit e13ec939e96b1 (fs: fix kernel_write prototype) on
-            # 9/1/2017.
+            # Determine the pos argument type, which was changed by commit
+            # e13ec939e96b ("fs: fix kernel_write prototype") in v4.14.
             #
             echo "$CONFTEST_PREAMBLE
             #include <linux/fs.h>
@@ -1803,9 +1799,8 @@ compile_test() {
 
         kernel_read_has_pointer_pos_arg)
             #
-            # Determine the pos argument type, which was changed by
-            # commit bdd1d2d3d251c (fs: fix kernel_read prototype) on
-            # 9/1/2017.
+            # Determine the pos argument type, which was changed by commit
+            # bdd1d2d3d251 ("fs: fix kernel_read prototype") in v4.14.
             #
             echo "$CONFTEST_PREAMBLE
             #include <linux/fs.h>
@@ -1830,8 +1825,8 @@ compile_test() {
             #
             # Determine if vm_insert_pfn_prot function is present
             #
-            # Added by commit 1745cbc5d0de ("mm: Add vm_insert_pfn_prot()") in
-            # v3.16.59
+            # Added by commit 1745cbc5d0de ("mm: Add vm_insert_pfn_prot()")
+            # in v4.6.
             #
             # Removed by commit f5e6d1d5f8f3 ("mm: introduce
             # vmf_insert_pfn_prot()") in v4.20.
@@ -2048,7 +2043,7 @@ compile_test() {
             # attached drivers") in v3.14 (2013-12-11)
             #
             # The commit 57bb1ee60340 ("drm: Compile out legacy chunks from
-            # struct drm_device") compiles out the legacy chunks like
+            # struct drm_device") in v5.11 compiles out the legacy chunks like
             # drm_driver::legacy_dev_list.
             #
             CODE="
@@ -2071,14 +2066,14 @@ compile_test() {
             #
             # Determine if jiffies_to_timespec() is present
             #
-            # removed by commit 751addac78b6
-            # ("y2038: remove obsolete jiffies conversion functions")
-            # in v5.6-rc1 (2019-12-13).
-        CODE="
-        #include <linux/jiffies.h>
-        void conftest_jiffies_to_timespec(void){
-            jiffies_to_timespec();
-        }"
+            # Removed by commit 751addac78b6 ("y2038: remove obsolete jiffies
+            # conversion functions") in v5.6.
+            #
+            CODE="
+            #include <linux/jiffies.h>
+            void conftest_jiffies_to_timespec(void){
+                jiffies_to_timespec();
+            }"
             compile_check_conftest "$CODE" "NV_JIFFIES_TO_TIMESPEC_PRESENT" "" "functions"
         ;;
 
@@ -2088,14 +2083,21 @@ compile_test() {
             #   drm_universal_plane_init()
             #   drm_crtc_init_with_planes()
             #   drm_encoder_init()
-            # have a 'name' argument, which was added by these commits:
-            #   drm_universal_plane_init:   2015-12-09  b0b3b7951114315d65398c27648705ca1c322faa
-            #   drm_crtc_init_with_planes:  2015-12-09  f98828769c8838f526703ef180b3088a714af2f9
-            #   drm_encoder_init:           2015-12-09  13a3d91f17a5f7ed2acd275d18b6acfdb131fb15
+            # have a 'name' argument.
+            #
+            # drm_universal_plane_init was updated by commit b0b3b7951114
+            # ("drm: Pass 'name' to drm_universal_plane_init()") in v4.5.
+            #
+            # drm_crtc_init_with_planes was updated by commit f98828769c88
+            # ("drm: Pass 'name' to drm_crtc_init_with_planes()") in v4.5.
             #
-            # Additionally determine whether drm_universal_plane_init() has a
-            # 'format_modifiers' argument, which was added by:
-            #   2017-07-23  e6fc3b68558e4c6d8d160b5daf2511b99afa8814
+            # drm_encoder_init was updated by commit 13a3d91f17a5 ("drm: Pass
+            # 'name' to drm_encoder_init()") in v4.5.
+            #
+            # Additionally, determine whether drm_universal_plane_init() has
+            # a 'format_modifiers' argument, which was added by commit
+            # e6fc3b68558e ("drm: Plumb modifiers through plane init") in
+            # v4.14.
             #
             CODE="
             #if defined(NV_DRM_DRMP_H_PRESENT)
@@ -2292,7 +2294,7 @@ compile_test() {
             # correction properties") in v4.6 (2016-03-08).
             #
             # Removed by commit f8ed34ac7b45 ("drm: drm_helper_crtc_enable_color_mgmt()
-            # => drm_crtc_enable_color_mgmt()") in v4.8-rc1 (2016-06-07).
+            # => drm_crtc_enable_color_mgmt()") in v4.8.
             #
             CODE="
             #include <drm/drm_crtc_helper.h>
@@ -2310,11 +2312,11 @@ compile_test() {
             # present.
             #
             # Added by commit f8ed34ac7b45 ("drm: drm_helper_crtc_enable_color_mgmt()
-            # => drm_crtc_enable_color_mgmt()") in v4.8-rc1 (2016-06-07), replacing
+            # => drm_crtc_enable_color_mgmt()") in v4.8, replacing
             # drm_helper_crtc_enable_color_mgmt().
             #
             # Moved to drm_color_mgmt.[ch] by commit f1e2f66ce2d9 ("drm: Extract
-            # drm_color_mgmt.[hc]") in v4.9-rc1 (2016-09-22)
+            # drm_color_mgmt.[hc]") in v4.9.
             #
             CODE="
             #if defined(NV_DRM_DRM_CRTC_H_PRESENT)
@@ -2341,8 +2343,7 @@ compile_test() {
             # Accidentally moved to drm_atomic_state_helper.[ch] by commit
             # 9ef8a9dc4b21 ("drm: Extract drm_atomic_state_helper.[ch]")
             # and moved back to drm_atomic_helper.[ch] by commit 1d8224e790c7
-            # ("drm: Fix up drm_atomic_state_helper.[hc] extraction") in
-            # v5.0-rc1
+            # ("drm: Fix up drm_atomic_state_helper.[hc] extraction") in v5.0.
             #
             # Removed by commit 6ca2ab8086af ("drm: automatic legacy gamma
             # support") in v5.12 (2020-12-15)
@@ -2406,8 +2407,8 @@ compile_test() {
             #
             # Added by commit 210647af897a ("PCI: Rename pci_remove_bus_device
             # to pci_stop_and_remove_bus_device") in v3.4 (2012-02-25) but
-            # aarch64 support was added by commit d1e6dc91b532 
-            # ("arm64: Add architectural support for PCI") in v3.18-rc1.
+            # aarch64 support was added by commit d1e6dc91b532 ("arm64: Add
+            # architectural support for PCI") in v3.18.
             #
             CODE="
             #include <linux/types.h>
@@ -2520,8 +2521,8 @@ compile_test() {
             #
             # Determine if the 'pci_dev' data type has a 'ats_enabled' member.
             #
-            # Added by commit d544d75ac96aa ("PCI: Embed ATS info directly
-            # into struct pci_dev") in v4.3-rc1 (2015-08-14)
+            # Added by commit d544d75ac96a ("PCI: Embed ATS info directly
+            # into struct pci_dev") in v4.3.
             #
             CODE="
             #include <linux/pci.h>
@@ -2552,9 +2553,9 @@ compile_test() {
             # commit 768ae309a961 ("mm: replace get_user_pages() write/force
             # parameters with gup_flags") in v4.9 (2016-10-13)
             #
-            # Removed vmas parameter from get_user_pages() by commit 7bbf9c8c99
+            # Removed vmas parameter from get_user_pages() by commit 54d020692b34
             # ("mm/gup: remove unused vmas parameter from get_user_pages()")
-            # in linux-next, expected in v6.5-rc1
+            # in v6.5.
             #
             # linux-4.4.168 cherry-picked commit 768ae309a961 without
             # c12d2da56d0e which is covered in Conftest #3.
@@ -2722,11 +2723,11 @@ compile_test() {
             #
             # get_user_pages_remote() removed 'tsk' parameter by
             # commit 64019a2e467a ("mm/gup: remove task_struct pointer for
-            # all gup code") in v5.9-rc1 (2020-08-11).
+            # all gup code") in v5.9.
             #
             # Removed vmas parameter from get_user_pages_remote() by commit
-            # a4bde14d549 ("mm/gup: remove vmas parameter from get_user_pages_remote()")
-            # in linux-next, expected in v6.5-rc1
+            # ca5e863233e8 ("mm/gup: remove vmas parameter from
+            # get_user_pages_remote()") in v6.5.
             #
 
             #
@@ -2925,15 +2926,14 @@ compile_test() {
             #
             # Determine if the function pin_user_pages() is present.
             # Presence of pin_user_pages() also implies the presence of
-            # unpin-user_page(). Both were added in the v5.6-rc1
+            # unpin-user_page().
             #
-            # pin_user_pages() was added by commit eddb1c228f7951d399240
-            # ("mm/gup: introduce pin_user_pages*() and FOLL_PIN") in
-            # v5.6-rc1 (2020-01-30)
+            # pin_user_pages() was added by commit eddb1c228f79 ("mm/gup:
+            # introduce pin_user_pages*() and FOLL_PIN") in v5.6.
             #
             # Removed vmas parameter from pin_user_pages() by commit
-            # 40896a02751("mm/gup: remove vmas parameter from pin_user_pages()")
-            # in linux-next, expected in v6.5-rc1
+            # 4c630f307455 ("mm/gup: remove vmas parameter from
+            # pin_user_pages()") in v6.5.
 
             set_pin_user_pages_defines () {
                 if [ "$1" = "" ]; then
@@ -2998,13 +2998,13 @@ compile_test() {
             # ("mm/gup: introduce pin_user_pages*() and FOLL_PIN")
             # in v5.6 (2020-01-30)
 
-            # pin_user_pages_remote() removed 'tsk' parameter by
-            # commit 64019a2e467a ("mm/gup: remove task_struct pointer for
-            # all gup code") in v5.9-rc1 (2020-08-11).
+            # pin_user_pages_remote() removed 'tsk' parameter by commit
+            # 64019a2e467a ("mm/gup: remove task_struct pointer for all gup
+            # code") in v5.9.
             #
             # Removed unused vmas parameter from pin_user_pages_remote() by
-            # commit 83bcc2e132 ("mm/gup: remove unused vmas parameter from
-            # pin_user_pages_remote()") in linux-next, expected in v6.5-rc1
+            # commit 0b295316b3a9 ("mm/gup: remove unused vmas parameter from
+            # pin_user_pages_remote()") in v6.5.
 
             #
             # This function sets the NV_PIN_USER_PAGES_REMOTE_* macros as per
@@ -3183,8 +3183,8 @@ compile_test() {
             #
             # Determine if enable_apicv boolean is exported by kernel.
             #
-            # Added by commit fdf513e37a3bd ("KVM: x86: Use common 'enable_apicv'
-            # variable for both APICv and AVIC")
+            # Added by commit fdf513e37a3b ("KVM: x86: Use common
+            # 'enable_apicv' variable for both APICv and AVIC") in v5.14.
             #
             CODE="
             $CONFTEST_PREAMBLE
@@ -4112,9 +4112,8 @@ compile_test() {
             # Determine if drm_connector_attach_vrr_capable_property and
             # drm_connector_set_vrr_capable_property is present
             #
-            # Added by commit ba1b0f6c73d4ea1390f0d5381f715ffa20c75f09 ("drm:
-            # Add vrr_capable property to the drm connector") in v5.0-rc1
-            # (2018-11-28)
+            # Added by commit ba1b0f6c73d4 ("drm: Add vrr_capable property to
+            # the drm connector") in v5.0.
             #
             CODE="
             #if defined(NV_DRM_DRM_CONNECTOR_H_PRESENT)
@@ -4424,16 +4423,21 @@ compile_test() {
             # with the logic of "functions" the presence of
             # *either*_alpha_property or _blend_mode_property would be enough
             # to cause NV_DRM_ALPHA_BLENDING_AVAILABLE to be defined.
+
+            # drm_plane_create_alpha_property was added by commit
+            # ae0e28265e21 ("drm/blend: Add a generic alpha property") in
+            # v4.18.
+            #
+            # drm_plane_create_blend_mode_property was added by commit
+            # a5ec8332d428 ("drm: Add per-plane pixel blend mode property")
+            # in v4.20.
             #
             CODE="
             #if defined(NV_DRM_DRM_BLEND_H_PRESENT)
             #include <drm/drm_blend.h>
             #endif
             void conftest_drm_alpha_blending_available(void) {
-                /* 2018-04-11 ae0e28265e216dad11d4cbde42fc15e92919af78 */
                 (void)drm_plane_create_alpha_property;
-
-                /* 2018-08-23 a5ec8332d4280500544e316f76c04a7adc02ce03 */
                 (void)drm_plane_create_blend_mode_property;
             }"
 
@@ -4444,10 +4448,10 @@ compile_test() {
             #
             # Determine if the DRM subsystem supports rotation.
             #
-            # drm_plane_create_rotation_property() was added on 2016-09-26 by
-            # d138dd3c0c70979215f3184cf36f95875e37932e (drm: Add support for
-            # optional per-plane rotation property) in linux kernel. Presence
-            # of it is sufficient to say that DRM subsystem support rotation.
+            # drm_plane_create_rotation_property() was added by commit
+            # d138dd3c0c70 ("drm: Add support for optional per-plane rotation
+            # property") in v4.10.  Presence of it is sufficient to say that
+            # DRM subsystem support rotation.
             #
             CODE="
             #if defined(NV_DRM_DRM_BLEND_H_PRESENT)
@@ -4466,8 +4470,8 @@ compile_test() {
             #
             # The DRIVER_PRIME flag was added by commit 3248877ea179 (drm:
             # base prime/dma-buf support (v5)) in v3.4 (2011-11-25) and is
-            # removed by commit 0424fdaf883a (drm/prime: Actually remove
-            # DRIVER_PRIME everywhere) on 2019-06-17.
+            # removed by commit 0424fdaf883a ("drm/prime: Actually remove
+            # DRIVER_PRIME everywhere") in v5.4.
             #
             # DRIVER_PRIME definition moved from drmP.h to drm_drv.h by
             # commit 85e634bce01a (drm: Extract drm_drv.h) in v4.10
@@ -4500,10 +4504,10 @@ compile_test() {
             #
             # drm_connector_for_each_possible_encoder() is added by commit
             # 83aefbb887b5 (drm: Add drm_connector_for_each_possible_encoder())
-            # in v4.19. The definition and prorotype is changed to take only
-            # two arguments connector and encoder, by commit 62afb4ad425a
-            # (drm/connector: Allow max possible encoders to attach to a
-            # connector) in v5.5rc1.
+            # in v4.19.  The definition and prototype is changed to take only
+            # two arguments connector and encoder by commit 62afb4ad425a
+            # ("drm/connector: Allow max possible encoders to attach to a
+            # connector") in v5.5.
             #
             echo "$CONFTEST_PREAMBLE
             #if defined(NV_DRM_DRMP_H_PRESENT)
@@ -4626,8 +4630,8 @@ compile_test() {
             #
             # Determine if the 'struct proc_ops' type is present.
             #
-            # Added by commit d56c0d45f0e2 ("proc: decouple proc from VFS with
-            # "struct proc_ops"") in 5.6-rc1
+            # Added by commit d56c0d45f0e2 ("proc: decouple proc from VFS
+            # with "struct proc_ops"") in v5.6.
             #
             CODE="
             #include <linux/proc_fs.h>
@@ -4685,8 +4689,8 @@ compile_test() {
             # Determine if 'drm_crtc_state' structure has a
             # 'vrr_enabled' field.
             #
-            # Added by commit 1398958cfd8d331342d657d37151791dd7256b40 ("drm:
-            # Add vrr_enabled property to drm CRTC") in v5.0-rc1 (2018-11-28)
+            # Added by commit 1398958cfd8d ("drm: Add vrr_enabled property to
+            # drm CRTC") in v5.0.
             #
             CODE="
             #if defined(NV_DRM_DRM_CRTC_H_PRESENT)
@@ -4707,11 +4711,11 @@ compile_test() {
             # Added by commit fb7fcc96a86cf ("timekeeping: Standardize on
             # ktime_get_*() naming") in 4.18 (2018-04-27)
             #
-        CODE="
-        #include <linux/ktime.h>
-        void conftest_ktime_get_raw_ts64(void){
-            ktime_get_raw_ts64();
-        }"
+            CODE="
+            #include <linux/ktime.h>
+            void conftest_ktime_get_raw_ts64(void){
+                ktime_get_raw_ts64();
+            }"
             compile_check_conftest "$CODE" "NV_KTIME_GET_RAW_TS64_PRESENT" "" "functions"
         ;;
 
@@ -4722,11 +4726,11 @@ compile_test() {
             # Added by commit d6d29896c665d ("timekeeping: Provide timespec64
             # based interfaces") in 3.17 (2014-07-16)
             #
-        CODE="
-        #include <linux/ktime.h>
-        void conftest_ktime_get_real_ts64(void){
-            ktime_get_real_ts64();
-        }"
+            CODE="
+            #include <linux/ktime.h>
+            void conftest_ktime_get_real_ts64(void){
+                ktime_get_real_ts64();
+            }"
             compile_check_conftest "$CODE" "NV_KTIME_GET_REAL_TS64_PRESENT" "" "functions"
         ;;
 
@@ -4746,8 +4750,9 @@ compile_test() {
             # -the "modifier[]" member of the AddFB2 ioctl's parameter
             #  structure.
             #
-            # All these were added by commit e3eb3250d84e (drm: add support for
-            # tiled/compressed/etc modifier in addfb2) in 4.1-rc1 (2015-02-05).
+            # All these were added by commit e3eb3250d84e ("drm: add support
+            # for tiled/compressed/etc modifier in addfb2") in v4.1.
+            #
             CODE="
             #include <drm/drm_mode.h>
             #include <drm/drm_fourcc.h>
@@ -4767,11 +4772,11 @@ compile_test() {
             # Added by commit 361a3bf00582 ("time64: Add time64.h header and
             # define struct timespec64") in 3.17 (2014-07-16)
             #
-        CODE="
-        #include <linux/time.h>
+            CODE="
+            #include <linux/time.h>
 
-        struct timespec64 ts64;
-        "
+            struct timespec64 ts64;
+            "
             compile_check_conftest "$CODE" "NV_TIMESPEC64_PRESENT" "" "types"
 
         ;;
@@ -4783,15 +4788,15 @@ compile_test() {
             # The third argument to __vmalloc, page protection
             # 'pgprot_t prot', was removed by commit 88dca4ca5a93
             # (mm: remove the pgprot argument to __vmalloc)
-            # in v5.8-rc1 (2020-06-01).
-        CODE="
-        #include <linux/vmalloc.h>
-
-        void conftest_vmalloc_has_pgprot_t_arg(void) {
-            pgprot_t prot;
-            (void)__vmalloc(0, 0, prot);
-        }"
+            # in v5.8.
+            #
+            CODE="
+            #include <linux/vmalloc.h>
 
+            void conftest_vmalloc_has_pgprot_t_arg(void) {
+                pgprot_t prot;
+                (void)__vmalloc(0, 0, prot);
+            }"
             compile_check_conftest "$CODE" "NV_VMALLOC_HAS_PGPROT_T_ARG" "" "types"
 
         ;;
@@ -4802,7 +4807,8 @@ compile_test() {
             #
             # Kernel commit da1c55f1b272 ("mmap locking API: rename mmap_sem
             # to mmap_lock") replaced the field 'mmap_sem' by 'mmap_lock'
-            # in v5.8-rc1 (2020-06-08).
+            # in v5.8.
+            #
             CODE="
             #include <linux/mm_types.h>
 
@@ -4892,9 +4898,9 @@ compile_test() {
         ;;
 
         pci_enable_atomic_ops_to_root)
-            # pci_enable_atomic_ops_to_root was added by
-            # commit 430a23689dea ("PCI: Add pci_enable_atomic_ops_to_root()")
-            # in v4.16-rc1 (2018-01-05)
+            #
+            # pci_enable_atomic_ops_to_root was added by commit 430a23689dea
+            # ("PCI: Add pci_enable_atomic_ops_to_root()") in v4.16.
             #
             CODE="
             #include <linux/pci.h>
@@ -4911,11 +4917,11 @@ compile_test() {
             # Added by commit a7c3e901a46ff54c016d040847eda598a9e3e653 ("mm:
             # introduce kv[mz]alloc helpers") in v4.12 (2017-05-08).
             #
-        CODE="
-        #include <linux/mm.h>
-        void conftest_kvmalloc(void){
-            kvmalloc();
-        }"
+            CODE="
+            #include <linux/mm.h>
+            void conftest_kvmalloc(void){
+                kvmalloc();
+            }"
             compile_check_conftest "$CODE" "NV_KVMALLOC_PRESENT" "" "functions"
 
         ;;
@@ -4924,12 +4930,11 @@ compile_test() {
             #
             # Determine if the function drm_gem_object_put_unlocked() is present.
             #
-            # In v5.9-rc1, commit 2f4dd13d4bb8 ("drm/gem: add
-            # drm_gem_object_put helper") removes drm_gem_object_put_unlocked()
-            # function and replace its definition by transient macro. Commit
-            # ab15d56e27be ("drm: remove transient
-            # drm_gem_object_put_unlocked()") finally removes
-            # drm_gem_object_put_unlocked() macro.
+            # Replaced with a transient macro by commit 2f4dd13d4bb8 ("drm/gem:
+            # add drm_gem_object_put helper") in v5.9.
+            #
+            # Finally removed by commit ab15d56e27be ("drm: remove transient
+            # drm_gem_object_put_unlocked()") in v5.9.
             #
             CODE="
             #if defined(NV_DRM_DRMP_H_PRESENT)
@@ -4952,7 +4957,7 @@ compile_test() {
             # field.
             #
             # Removed by commit 0425662fdf05 ("drm: Nuke mode->vrefresh") in
-            # v5.9-rc1.
+            # v5.9.
             #
             CODE="
             #include <drm/drm_modes.h>
@@ -4970,7 +4975,7 @@ compile_test() {
             # Determine if drm_driver::master_set() returns integer value
             #
             # Changed to void by commit 907f53200f98 ("drm: vmwgfx: remove
-            # drm_driver::master_set() return type") in v5.9-rc1.
+            # drm_driver::master_set() return type") in v5.9.
             #
             CODE="
             #if defined(NV_DRM_DRMP_H_PRESENT)
@@ -4996,7 +5001,7 @@ compile_test() {
             # function pointer.
             #
             # drm_driver::gem_free_object is removed by commit 1a9458aeb8eb
-            # ("drm: remove drm_driver::gem_free_object") in v5.9-rc1.
+            # ("drm: remove drm_driver::gem_free_object") in v5.9.
             #
             CODE="
             #if defined(NV_DRM_DRMP_H_PRESENT)
@@ -5019,7 +5024,7 @@ compile_test() {
             # Determine if vga_tryget() is present
             #
             # vga_tryget() was removed by commit f369bc3f9096 ("vgaarb: mark
-            # vga_tryget static") in v5.9-rc1 (2020-08-01).
+            # vga_tryget static") in v5.9.
             #
             CODE="
             #include <linux/vgaarb.h>
@@ -5036,7 +5041,7 @@ compile_test() {
             #
             # pci_channel_state was removed by commit 16d79cd4e23b ("PCI: Use
             # 'pci_channel_state_t' instead of 'enum pci_channel_state'") in
-            # v5.9-rc1 (2020-07-02).
+            # v5.9.
             #
             CODE="
             #include <linux/pci.h>
@@ -5052,7 +5057,8 @@ compile_test() {
             # Determine if 'cc_platform_has()' is present.
             #
             # Added by commit aa5a461171f9 ("x86/sev: Add an x86 version of
-            # cc_platform_has()") in v5.15.3 (2021-10-04)
+            # cc_platform_has()") in v5.16.
+            #
             CODE="
             #if defined(NV_LINUX_CC_PLATFORM_H_PRESENT)
             #include <linux/cc_platform.h>
@@ -5069,8 +5075,9 @@ compile_test() {
             #
             # Determine if drm_prime_pages_to_sg() has 'dev' argument.
             #
-            # drm_prime_pages_to_sg() is updated to take 'dev' argument by commit
-            # 707d561f77b5 ("drm: allow limiting the scatter list size.").
+            # drm_prime_pages_to_sg() is updated to take 'dev' argument by
+            # commit 707d561f77b5 ("drm: allow limiting the scatter list
+            # size.") in v5.10.
             #
             CODE="
             #if defined(NV_DRM_DRMP_H_PRESENT)
@@ -5094,9 +5101,9 @@ compile_test() {
             # Determine if drm_driver structure has the GEM and PRIME callback
             # function pointers.
             #
-            # The GEM and PRIME callback are removed from drm_driver
-            # structure, by commit d693def4fd1c ("drm: Remove obsolete GEM and
-            # PRIME callbacks from struct drm_driver").
+            # The GEM and PRIME callbacks are removed from drm_driver
+            # structure by commit d693def4fd1c ("drm: Remove obsolete GEM and
+            # PRIME callbacks from struct drm_driver") in v5.11.
             #
             CODE="
             #if defined(NV_DRM_DRMP_H_PRESENT)
@@ -5125,8 +5132,8 @@ compile_test() {
             # Determine if drm_crtc_helper_funcs::atomic_check takes 'state'
             # argument of 'struct drm_atomic_state' type.
             #
-            # The commit 29b77ad7b9ca ("drm/atomic: Pass the full state to CRTC
-            # atomic_check") passed the full atomic state to
+            # Commit 29b77ad7b9ca ("drm/atomic: Pass the full state to CRTC
+            # atomic_check") in v5.11 passed the full atomic state to
             # drm_crtc_helper_funcs::atomic_check()
             #
             # To test the signature of drm_crtc_helper_funcs::atomic_check(),
@@ -5162,9 +5169,9 @@ compile_test() {
             # Determine if drm_gem_object_funcs::vmap takes 'map'
             # argument of 'struct dma_buf_map' type.
             #
-            # The commit 49a3f51dfeee ("drm/gem: Use struct dma_buf_map in GEM
-            # vmap ops and convert GEM backends") update
-            # drm_gem_object_funcs::vmap to take 'map' argument.
+            # drm_gem_object_funcs::vmap is updated to take 'map' argument by
+            # commit 49a3f51dfeee ("drm/gem: Use struct dma_buf_map in GEM
+            # vmap ops and convert GEM backends") in v5.11.
             #
             # Note that the 'map' argument type is changed from 'struct dma_buf_map'
             # to 'struct iosys_map' by commit 7938f4218168 ("dma-buf-map: Rename
@@ -5185,7 +5192,7 @@ compile_test() {
             # Determine if seq_read_iter() is present
             #
             # seq_read_iter() was added by commit d4d50710a8b4 ("seq_file:
-            # add seq_read_iter") in v5.10-rc1 (2020-11-04).
+            # add seq_read_iter") in v5.10.
             #
             CODE="
             #include <linux/seq_file.h>
@@ -5203,7 +5210,7 @@ compile_test() {
             #
             # The commit 07f4f97d7b4b ("vga_switcheroo: Use device link for HDA
             # controller") has moved 'PCI_CLASS_MULTIMEDIA_HD_AUDIO' macro from
-            # <sound/hdaudio.h> to <linux/pci_ids.h> in v4.17-rc1 (2018-03-03).
+            # <sound/hdaudio.h> to <linux/pci_ids.h> in v4.17.
             #
             CODE="
             #include <linux/pci_ids.h>
@@ -5223,6 +5230,9 @@ compile_test() {
             # by commit 233eb0bf3b94 ("mm: remove follow_pfn")
             # from linux-next 233eb0bf3b94.
             #
+            # Note: this commit never made it to the linux kernel, so
+            # unsafe_follow_pfn() never existed.
+            #
             CODE="
             #include <linux/mm.h>
             void conftest_follow_pfn(void) {
@@ -5236,8 +5246,8 @@ compile_test() {
             # Determine if drm_plane_helper_funcs::atomic_check takes 'state'
             # argument of 'struct drm_atomic_state' type.
             #
-            # The commit 7c11b99a8e58 ("drm/atomic: Pass the full state to
-            # planes atomic_check") passed the full atomic state to
+            # Commit 7c11b99a8e58 ("drm/atomic: Pass the full state to planes
+            # atomic_check") in v5.13 passes the full atomic state to
             # drm_plane_helper_funcs::atomic_check()
             #
             # To test the signature of drm_plane_helper_funcs::atomic_check(),
@@ -5301,7 +5311,7 @@ compile_test() {
             # Determine if the add_memory_driver_managed function is present
             #
             # Added by commit 7b7b27214bba ("mm/memory_hotplug: introduce
-            # add_memory_driver_managed()") in v5.8-rc1 (2020-06-05)
+            # add_memory_driver_managed()") in v5.8.
             #
             # Before commit 3a0aaefe4134 ("mm/memory_hotplug: guard more
             # declarations by CONFIG_MEMORY_HOTPLUG") in v5.10, the
@@ -5322,8 +5332,8 @@ compile_test() {
             #
             # Check if add_memory_driver_managed() has mhp_flags arg.
             #
-            # Added by commit b6117199787c ("mm/memory_hotplug: prepare passing flags to
-            # add_memory() and friends") in v5.10-rc1 (2020-10-16)
+            # Added by commit b6117199787c ("mm/memory_hotplug: prepare
+            # passing flags to add_memory() and friends") in v5.10.
             #
             CODE="
             #include <linux/memory_hotplug.h>
@@ -5340,8 +5350,8 @@ compile_test() {
             #
             # Check if remove_memory() has nid parameter.
             #
-            # Removed by commit e1c158e4956612e7 ("mm/memory_hotplug: remove nid
-            # parameter from remove_memory() and friends") in v5.15-rc1 (2021-09-09)
+            # Removed by commit e1c158e49566 ("mm/memory_hotplug: remove nid
+            # parameter from remove_memory() and friends") in v5.15.
             #
             CODE="
             #include <linux/memory_hotplug.h>
@@ -5356,8 +5366,8 @@ compile_test() {
             #
             # Determine if the offline_and_remove_memory function is present.
             #
-            # Added by commit 08b3acd7a68fc179 ("mm/memory_hotplug: Introduce
-            # offline_and_remove_memory()") in v5.8-rc1 (2020-06-05)
+            # Added by commit 08b3acd7a68f ("mm/memory_hotplug: Introduce
+            # offline_and_remove_memory()") in v5.8.
             #
             CODE="
             #include <linux/memory_hotplug.h>
@@ -5372,8 +5382,8 @@ compile_test() {
             #
             # Determine if the device_property_read_u64 function is present
             #
-            # Added by commit b31384fa5de37a1 ("Driver core: Unified device
-            # properties interface for platform firmware") in v3.19-rc1 (2014-11-05)
+            # Added by commit b31384fa5de3 ("Driver core: Unified device
+            # properties interface for platform firmware") in v3.19.
             #
             CODE="
             #include <linux/acpi.h>
@@ -5388,8 +5398,12 @@ compile_test() {
             #
             # Determine if of_property_count_elems_of_size is present
             #
-            # Added by commit 1df09bcof (" Move OF property and graph API from
-            # base.c to property.c"
+            # Added by commit ad54a0cfbeb4 ("of: add functions to count
+            # number of elements in a property") in v3.15.
+            #
+            # Moved from base.c to property.c by commit 1df09bc66f9b ("of:
+            # Move OF property and graph API from base.c to property.c") in
+            # v4.13.
             #
             # Test if linux/of.h header file inclusion is successful or not,
             # depending on that check, for of_property_count_elems_of_size
@@ -5420,8 +5434,12 @@ compile_test() {
             #
             # Determine if of_property_read_variable_u8_array is present
             #
-            # Added by commit 1df09bcof (" Move OF property and graph API from
-            # base.c to property.c"
+            # Added by commit a67e9472da42 ("of: Add array read functions
+            # with min/max size limits") in v4.9.
+            #
+            # Moved from base.c to property.c by commit 1df09bc66f9b ("of:
+            # Move OF property and graph API from base.c to property.c") in
+            # v4.13.
             #
             # Test if linux/of.h header file inclusion is successful or not,
             # depending on that, check for of_property_read_variable_u8_array
@@ -5452,8 +5470,15 @@ compile_test() {
             #
             # Determine if of_property_read_variable_u32_array is present
             #
-            # Added by commit 1df09bcof (" Move OF property and graph API from
-            # base.c to property.c"
+            # Added by commit a67e9472da42 ("of: Add array read functions
+            # with min/max size limits") in v4.9.
+            #
+            # Moved from base.c to property.c by commit 1df09bc66f9b ("of:
+            # Move OF property and graph API from base.c to property.c") in
+            # v4.13.
+            #
+            # Note: this can probably be combined with the
+            # of_property_read_variable_u8_array conftest above.
             #
             # Test if linux/of.h header file inclusion is successful or not,
             # depending on that, check for of_property_read_variable_u32_array
@@ -5501,8 +5526,8 @@ compile_test() {
             #
             # Determine if devm_of_platform_populate() function is present
             #
-            # Added by commit 38b0b21of (add devm_ functions for populate and
-            # depopulate")
+            # Added by commit 38b0b219fbe8 ("of: add devm_ functions for
+            # populate and depopulate") in v4.12.
             #
             CODE="
             #if defined(NV_LINUX_OF_PLATFORM_H_PRESENT)
@@ -5520,8 +5545,13 @@ compile_test() {
             #
             # Determine if of_dma_configure() function is present
             #
-            # Added by commit 591c1eeof ("configure the platform device
-            # dma parameters")
+            # Added by commit 591c1ee465ce ("of: configure the platform
+            # device dma parameters") in v3.16.  However, it was a static,
+            # non-exported function at that time.
+            #
+            # It was moved from platform.c to device.c and made public by
+            # commit 1f5c69aa51f9 ("of: Move of_dma_configure() to device.c
+            # to help re-use") in v4.1.
             #
             CODE="
             #if defined(NV_LINUX_OF_DEVICE_H_PRESENT)
@@ -5540,8 +5570,8 @@ compile_test() {
             #
             # Determine if icc_get() function is present
             #
-            # Added by commit 11f1cec ("interconnect: Add generic on-chip
-            # interconnect API")
+            # Added by commit 11f1ceca7031 ("interconnect: Add generic
+            # on-chip interconnect API") in v5.1.
             #
             CODE="
             #if defined(NV_LINUX_INTERCONNECT_H_PRESENT)
@@ -5560,8 +5590,8 @@ compile_test() {
             #
             # Determine if icc_set_bw() function is present
             #
-            # Added by commit 11f1cec ("interconnect: Add generic on-chip
-            # interconnect API")
+            # Added by commit 11f1ceca7031 ("interconnect: Add generic
+            # on-chip interconnect API") in v5.1.
             #
             CODE="
             #if defined(NV_LINUX_INTERCONNECT_H_PRESENT)
@@ -5580,8 +5610,8 @@ compile_test() {
             #
             # Determine if icc_put() function is present
             #
-            # Added by commit 11f1cec ("interconnect: Add generic on-chip
-            # interconnect API")
+            # Added by commit 11f1ceca7031 ("interconnect: Add generic
+            # on-chip interconnect API") in v5.1.
             #
             CODE="
             #if defined(NV_LINUX_INTERCONNECT_H_PRESENT)
@@ -5600,7 +5630,8 @@ compile_test() {
             #
             # Determine if i2c_new_client_device() function is present
             #
-            # Added by commit 390fd04i2c ("remove deprecated i2c_new_device API")
+            # Added by commit 390fd0475af5 ("i2c: remove deprecated
+            # i2c_new_device API") in v5.8.
             #
             CODE="
             #include <linux/i2c.h>
@@ -5617,7 +5648,8 @@ compile_test() {
             #
             # Determine if i2c_unregister_device() function is present
             #
-            # Added by commit 9c1600ei2c ("Add i2c_board_info and i2c_new_device()")
+            # Added by commit 9c1600eda42e ("i2c: Add i2c_board_info and
+            # i2c_new_device()") in v2.6.22.
             #
             CODE="
             #include <linux/i2c.h>
@@ -5634,8 +5666,8 @@ compile_test() {
             #
             # Determine if of_get_named_gpio() function is present
             #
-            # Added by commit a6b0919 ("of/gpio: Add new method for getting gpios
-            # under different property names")
+            # Added by commit a6b0919140b4 ("of/gpio: Add new method for
+            # getting gpios under different property names") in v3.1.
             #
             CODE="
             #if defined(NV_LINUX_OF_GPIO_H_PRESENT)
@@ -5654,7 +5686,8 @@ compile_test() {
             #
             # Determine if devm_gpio_request_one() function is present
             #
-            # Added by commit 09d71ff (gpiolib: Implement devm_gpio_request_one()")
+            # Added by commit 09d71ff19404 ("gpiolib: Implement
+            # devm_gpio_request_one()") in v3.5.
             #
             CODE="
             #if defined(NV_LINUX_GPIO_H_PRESENT)
@@ -5673,7 +5706,8 @@ compile_test() {
             #
             # Determine if gpio_direction_input() function is present
             #
-            # Added by commit c7caf86 (gpio: remove gpio_ensure_requested()")
+            # Added by commit c7caf86823c7 ("gpio: remove
+            # gpio_ensure_requested()") in v3.17.
             #
             CODE="
             #if defined(NV_LINUX_GPIO_H_PRESENT)
@@ -5692,7 +5726,8 @@ compile_test() {
             #
             # Determine if gpio_direction_output() function is present
             #
-            # Added by commit c7caf86 (gpio: remove gpio_ensure_requested()")
+            # Added by commit c7caf86823c7 ("gpio: remove
+            # gpio_ensure_requested()") in v3.17.
             #
             CODE="
             #if defined(NV_LINUX_GPIO_H_PRESENT)
@@ -5711,8 +5746,8 @@ compile_test() {
             #
             # Determine if gpio_get_value() function is present
             #
-            # Added by commit 7563bbf ("gpiolib/arches: Centralise bolierplate
-            # asm/gpio.h")
+            # Added by commit 7563bbf89d06 ("gpiolib/arches: Centralise
+            # bolierplate asm/gpio.h") in v3.5.
             #
             CODE="
             #if defined(NV_LINUX_GPIO_H_PRESENT)
@@ -5731,8 +5766,8 @@ compile_test() {
             #
             # Determine if gpio_set_value() function is present
             #
-            # Added by commit 7563bbf ("gpiolib/arches: Centralise bolierplate
-            # asm/gpio.h")
+            # Added by commit 7563bbf89d06 ("gpiolib/arches: Centralise
+            # bolierplate asm/gpio.h") in v3.5.
             #
             CODE="
             #if defined(NV_LINUX_GPIO_H_PRESENT)
@@ -5751,8 +5786,8 @@ compile_test() {
             #
             # Determine if gpio_to_irq() function is present
             #
-            # Added by commit 7563bbf ("gpiolib/arches: Centralise bolierplate
-            # asm/gpio.h")
+            # Added by commit 7563bbf89d06 ("gpiolib/arches: Centralise
+            # bolierplate asm/gpio.h") in v3.5.
             #
             CODE="
             #if defined(NV_LINUX_GPIO_H_PRESENT)
@@ -5771,10 +5806,9 @@ compile_test() {
             #
             # Determine if migrate_vma structure has flags
             #
-            # flags were added to struct migrate_vma by commit
-            # 5143192cd410c4fc83be09a2e73423765aee072b ("mm/migrate: add a flags
-            # parameter to_migrate_vma) in v5.9.
-            # (2020-07-28).
+            # Added by commit 5143192cd410 ("mm/migrate: add a flags
+            # parameter to migrate_vma") in v5.9.
+            #
             CODE="
             #include <linux/migrate.h>
             int conftest_migrate_vma_added_flags(void) {
@@ -5789,7 +5823,7 @@ compile_test() {
             # Determine if the 'drm_device' structure has a 'pdev' field.
             #
             # Removed by commit b347e04452ff ("drm: Remove pdev field from
-            # struct drm_device") in v5.14-rc1.
+            # struct drm_device") in v5.14.
             #
             CODE="
             #if defined(NV_DRM_DRMP_H_PRESENT)
@@ -5843,9 +5877,9 @@ compile_test() {
             #
             # Determine if ioasid_get() function is present
             #
-            # ioasid_get() function was added by commit
-            # cb4789b0d19ff231ce9f73376a023341300aed96 (iommu/ioasid: Add ioasidreferences) in v5.11.
-            # (2020-11-23).
+            # Added by commit cb4789b0d19f ("iommu/ioasid: Add ioasid
+            # references") in v5.11.
+            #
             CODE="
             #if defined(NV_LINUX_IOASID_H_PRESENT)
             #include <linux/ioasid.h>
@@ -5882,9 +5916,8 @@ compile_test() {
             #
             # Determine if the 'drm_crtc_state' structure has 'no_vblank'.
             #
-            # drm_crtc_state::no_vblank was added by commit b25c60af7a877
-            # ("drm/crtc: Add a generic infrastructure to fake VBLANK events")
-            # in 4.18.0-rc3 (2018-07-03).
+            # Added by commit b25c60af7a87 ("drm/crtc: Add a generic
+            # infrastructure to fake VBLANK events") in v4.19.
             #
             CODE="
             #include <drm/drm_crtc.h>
@@ -5904,7 +5937,7 @@ compile_test() {
             # an 'allow_fb_modifiers' field in the 'drm_mode_config' structure,
             # is added by commit e3eb3250d84e ("drm: add support for
             # tiled/compressed/etc modifier in addfb2") in v4.1, and removed by
-            # commit 3d082157a242 ("drm: remove allow_fb_modifiers") in v5.18-rc1.
+            # commit 3d082157a242 ("drm: remove allow_fb_modifiers") in v5.18.
             #
             # The 'struct drm_mode_config' definition, is moved to
             # drm_mode_config.h file by commit 28575f165d36 ("drm: Extract
@@ -5942,9 +5975,8 @@ compile_test() {
             #
             # Determine if drm_mode.h has 'hdr_output_metadata' structure.
             #
-            # struct hdr_output_metadata was added by commit fbb5d0353c62d
-            # ("drm: Add HDR source metadata property") in 5.1.0-rc5
-            # (2019-05-16)
+            # Added by commit fbb5d0353c62 ("drm: Add HDR source metadata
+            # property") in v5.3.
             #
             CODE="
             #include <drm/drm_mode.h>
@@ -5971,9 +6003,8 @@ compile_test() {
             #
             # Determine if the platform_irq_count() function is present
             #
-            # platform_irq_count was added by commit
-            # 4b83555d5098e73cf2c5ca7f86c17ca0ba3b968e ("driver-core: platform: Add platform_irq_count()")
-            # in 4.5-rc1 (2016-01-07)
+            # Added by commit 4b83555d5098 ("driver-core: platform: Add
+            # platform_irq_count()") in v4.5.
             #
             CODE="
             #include <linux/platform_device.h>
@@ -5987,7 +6018,8 @@ compile_test() {
             #
             # Determine if devm_clk_bulk_get_all() function is present
             #
-            # Added by commit f08c2e286 ("clk: add managed version of clk_bulk_get_all")
+            # Added by commit f08c2e2865f6 ("clk: add managed version of
+            # clk_bulk_get_all") in v4.20.
             #
             CODE="
             #if defined(NV_LINUX_CLK_H_PRESENT)
@@ -6054,7 +6086,7 @@ compile_test() {
             # dma_resv_add_excl_fence() and dma_resv_add_shared_fence() were
             # removed and replaced with dma_resv_add_fence() by commit
             # 73511edf8b19 ("dma-buf: specify usage while adding fences to
-            # dma_resv obj v7") in linux-next, expected in v5.19-rc1.
+            # dma_resv obj v7") in v5.19.
             #
             CODE="
             #if defined(NV_LINUX_DMA_RESV_H_PRESENT)
@@ -6074,7 +6106,7 @@ compile_test() {
             # dma_resv_reserve_shared() was removed and replaced with
             # dma_resv_reserve_fences() by commit c8d4c18bfbc4
             # ("dma-buf/drivers: make reserving a shared slot mandatory v4") in
-            # linux-next, expected in v5.19-rc1.
+            # v5.19.
             #
             CODE="
             #if defined(NV_LINUX_DMA_RESV_H_PRESENT)
@@ -6094,8 +6126,7 @@ compile_test() {
             #
             # reservation_object_reserve_shared() function prototype was updated
             # to take 'num_fences' argument by commit ca05359f1e64 ("dma-buf:
-            # allow reserving more than one shared fence slot") in v4.21-rc1
-            # (2018-12-14).
+            # allow reserving more than one shared fence slot") in v5.0.
             #
             CODE="
             #include <linux/reservation.h>
@@ -6112,9 +6143,8 @@ compile_test() {
             #
             # Determine if the __get_task_ioprio() function is present.
             #
-            # __get_task_ioprio was added by commit 893e5d32d583
-            # ("block: Generalize get_current_ioprio() for any task") for
-            # v5.20 linux-next (2022-06-23).
+            # Added by commit 893e5d32d583 ("block: Generalize
+            # get_current_ioprio() for any task") in v6.0.
             #
             CODE="
             #include <linux/ioprio.h>
@@ -6129,9 +6159,8 @@ compile_test() {
             #
             # Determine if 'num_registered_fb' variable is present.
             #
-            # 'num_registered_fb' was removed by commit 5727dcfd8486
-            # ("fbdev: Make registered_fb[] private to fbmem.c") for
-            # v5.20 linux-next (2022-07-27).
+            # Removed by commit 5727dcfd8486 ("fbdev: Make registered_fb[]
+            # private to fbmem.c") in v6.1.
             #
             CODE="
             #include <linux/fb.h>
@@ -6277,9 +6306,8 @@ compile_test() {
             #
             # Determine if 'struct drm_connector' has an 'override_edid' member.
             #
-            # Removed by commit 90b575f52c6ab ("drm/edid: detach debugfs EDID
-            # override from EDID property update") in linux-next, expected in
-            # v6.2-rc1.
+            # Removed by commit 90b575f52c6a ("drm/edid: detach debugfs EDID
+            # override from EDID property update") in v6.2.
             #
             CODE="
             #if defined(NV_DRM_DRM_CRTC_H_PRESENT)
@@ -6321,10 +6349,9 @@ compile_test() {
             # Determine if the 'vm_area_struct' structure has
             # const 'vm_flags'.
             #
-            # A union of '__vm_flags' and 'const vm_flags' was added 
-            # by commit bc292ab00f6c ("mm: introduce vma->vm_flags
-            # wrapper functions") in mm-stable branch (2023-02-09)
-            # of the akpm/mm maintainer tree.
+            # A union of '__vm_flags' and 'const vm_flags' was added by
+            # commit bc292ab00f6c ("mm: introduce vma->vm_flags wrapper
+            # functions") in v6.3.
             #
             CODE="
             #include <linux/mm_types.h>
@@ -6340,8 +6367,8 @@ compile_test() {
             # Determine if the 'drm_driver' structure has a 'dumb_destroy'
             # function pointer.
             #
-            # Removed by commit 96a7b60f6ddb2 ("drm: remove dumb_destroy
-            # callback") in v6.3 linux-next (2023-02-10).
+            # Removed by commit 96a7b60f6ddb ("drm: remove dumb_destroy
+            # callback") in v6.4.
             #
             CODE="
             #if defined(NV_DRM_DRMP_H_PRESENT)
@@ -6363,9 +6390,8 @@ compile_test() {
             #
             # Check if memory_failure() has trapno parameter.
             #
-            # trapno argument was removed by commit
-            # 83b57531c58f4173d1c0d0b2c0bc88c853c32ea5 ("mm/memory_failure:
-            # Remove unused trapno from memory_failure") in v4.15.0 (2017-7-9)
+            # Removed by commit 83b57531c58f ("mm/memory_failure: Remove
+            # unused trapno from memory_failure") in v4.16.
             #
             CODE="
             #include <linux/mm.h>
@@ -6382,9 +6408,8 @@ compile_test() {
             #
             # Check if memory_failure() flag MF_SW_SIMULATED is defined.
             #
-            # MF_SW_SIMULATED was added by commit
-            # 67f22ba7750f940bcd7e1b12720896c505c2d63f ("mm/hwpoison:
-            # fix unpoison_memory()") in v5.19.0-rc2 (2022-6-16)
+            # Added by commit 67f22ba7750f ("mm/memory-failure: disable
+            # unpoison once hw error happens") in v5.19.
             #
             CODE="
             #include <linux/mm.h>
@@ -6564,10 +6589,12 @@ compile_test() {
         ;;
 
         # When adding a new conftest entry, please use the correct format for
-        # specifying the relevant upstream Linux kernel commit.
+        # specifying the relevant upstream Linux kernel commit.  Please
+        # avoid specifying -rc kernels, and only use SHAs that actually exist
+        # in the upstream Linux kernel git repository.
         #
-        # <function> was added|removed|etc by commit <sha> ("<commit message")
-        # in <kernel-version> (<commit date>).
+        # Added|Removed|etc by commit <short-sha> ("<commit message") in
+        # <kernel-version>.
 
         *)
             # Unknown test name given
-- 
2.39.5

